docs/kratos/hooks/01_configure-hooks.mdx

```diff
@@ -0,0 +1,373 @@
+
+Adding hooks to self-service flows can help you shape your customers' user experience and increase the security of your systems.
+
+## When can you use hooks?
+
+You can use hooks at specific points of select self-service flows:
+
+- `before` login: The hook runs when the user starts the login flow.
+- `after` login: The hook runs when the user is successfully authenticated, before the system issues an Ory Session.
+- `before` registration: The hook runs when a registration flow starts.
+- `after` registration: The hook runs when a user successfully registers in the system.
+- `after` recovery: The hook runs when the user successfully recovers their password.
+- `after` settings: The hook runs when the user successfully changes their account settings.
+- `after` verification: The hook runs when the user verifies their account.
+
+### Hooks for specific methods
+
+The login and registration self-service flows support multiple authentication methods:
+
+| Authentication method | Description                                                                                    |
+| :-------------------- | :--------------------------------------------------------------------------------------------- |
+| `password`            | Sign-in and sign-up with username and password combo.                                          |
+| `oidc`                | Sign-in and sign-up through OIDC-compliant OAuth2 identity providers.                          |
+| `totp`                | Sign-in with a TOTP code from apps such as Google Authenticator.                               |
+| `webauthn`            | Sign-in with WebAuthn-compatible factors (FaceID, YubiKey) or paswordless sign-up and sign-in. |
+| `lookup_secret`       | Sign-in with recovery codes.                                                                   |
+
+:::tip
+
+You can use the authentication methods to further specify when you want to run hooks. For example, you can run a hook for the
+login flow with the `password` method and not run any hooks for logins with the `oidc` method.
+
+:::
+
+## Available hooks
+
+There are 4 hooks you can use to extend self-service user flows. Note that some hooks can only be used for specific flows and
+methods.
+
+| Hook                       | Description                                                                                       | Constraints                                                                                              |
+| :------------------------- | :------------------------------------------------------------------------------------------------ | :------------------------------------------------------------------------------------------------------- |
+| `session`                  | Signs in the user immediately after they create an account.                                       | For use only with the `after` registration flow. To use it, you must define secrets for secret rotation. |
```

Comment:
Not sure to understand what `secret` refers to here

Reply:
```suggestion
| `session`                  | Signs in the user immediately after they create an account.                                       | For use only with the `after` registration flow. To use it, you must define [secrets](https://docs-6vm1c68lu-ory.vercel.app/docs/kratos/hooks/configure-hooks#session-hook-configuration) for secret rotation. |
```

Reply:
There's a whole paragraph that explains that ;)

Reply:
Added a link